packages/react-core/src/components/Form/__tests__/FormGroup.test.tsx

```diff
@@ -203,4 +203,52 @@ describe('FormGroup', () => {
     );
     expect(asFragment()).toMatchSnapshot();
   });
+
+  describe('With multiple inputs per group', () => {
+    test('should render with group role when multipleInputs.isRadioGroup is false', () => {
+      render(
+        <FormGroup label="label" fieldId="group-role" multipleInputs={{ isRadioGroup: false }}>
+          <input />
+          <input />
+        </FormGroup>
+      );
+
+      expect(screen.getByRole('group')).toBeInTheDocument();
+    });
+
+    test('should render with radiogroup role when multipleInputs.isRadioGroup is true', () => {
+      render(
+        <FormGroup label="label" fieldId="radiogroup-role" multipleInputs={{ isRadioGroup: true }}>
+          <input />
+          <input />
+        </FormGroup>
+      );
+
+      expect(screen.getByRole('radiogroup')).toBeInTheDocument();
+    });
+
+    test('should have accessible name when multipleInputs is passed in', () => {
+      render(
+        <FormGroup label="label" fieldId="accessible-name" multipleInputs={{ isRadioGroup: false }}>
+          <input />
+          <input />
+        </FormGroup>
+      );
+
+      expect(screen.getByRole('group')).toHaveAccessibleName('label');
+    });
+
+    test('should have correct element for group label when multipleInputs is passed in', () => {
+      render(
+        <FormGroup label="label" fieldId="correct-label-element" multipleInputs={{ isRadioGroup: false }}>
+          <input />
+          <input />
+        </FormGroup>
+      );
+
+      const labelElement = screen.getByRole('group').querySelector('.pf-c-form__label');
+
+      expect(labelElement.tagName).toBe('SPAN');
```

Comment:
This feels very non-RTL to me, but I'll have to think about if there's a way to test this that would be more in line with a real user experience and circle back to you. There very well might not be.

Reply:
Do you think testing against a snapshot would make more sense here? Once unit tests are rewritten to align better with our current intentions, this would ideally only result in 2 snapshots for the component.

Another possibility *might* be testing user events. If there's a form group with a single input, clicking the label should place focus on the input; if there's a form group with multiple inputs, however, clicking the span won't place focus anywhere.

Reply:
> Another possibility might be testing user events. If there's a form group with a single input, clicking the label should place focus on the input; if there's a form group with multiple inputs, however, clicking the span won't place focus anywhere.

That seems like it aligns super well with the actual difference in UX and is the way I would go, awesome idea!